languages/rust/exercises/concept/vec/.meta/example.rs

```diff
@@ -0,0 +1,24 @@
+/// return a list of 5 languages to learn
+pub fn languages() -> Vec<String> {
+    vec![
+        "Rust".to_string(),
+        "JavaScript".to_string(),
+        "Elixir".to_string(),
+        "Lua".to_string(),
+        "Scheme".to_string(),
+    ]
+}
+/// Gather your top languages to learn
+pub fn grab(which: usize) -> Option<String> {
+    if which == languages().len() {
+        languages().pop()
+    } else {
+        languages().get(which).cloned()
+    }
+}
```

Comment:
This function has kind of weird semantics. As written, we could assert

```rust
assert_eq!(grab(4), Some("Scheme".into());
assert_eq!(grab(5), Some("Scheme".into());
assert_eq!(grab(6), None);
```

Reply:
Removed pop semantics as I had a better idea on how to introduce that.